lib/render/FlowRenderer.cpp

```diff
@@ -110,66 +113,98 @@ FlowRenderer::_initializeGL()
     return 0;
 }
 
-int
-FlowRenderer::_paintGL( bool fast )
+
+int FlowRenderer::_paintGL( bool fast )
 {
     FlowParams* params = dynamic_cast<FlowParams*>( GetActiveParams() );
     int rv;     // return value
 
     _velocityField.DefaultZ = Renderer::GetDefaultZ( _dataMgr, params->GetCurrentTimestep() );
 
-    if( params->GetNeedFlowlineOutput() )
-    {
+    if( params->GetNeedFlowlineOutput() ) {
```

Comment:
The _paintGL method is getting awfully long and complex. Can you make this 'if' clause into a helper function to improve readability?

Reply:
> Looks good, Sam. A couple of in-line comments and one question: Is there a reason to prefer the 'trailing return type' syntax:
> 
> `auto f(int x, int y) -> int {}`
> 
> over the conventional:
> 
> `int f(int x, int y)`
> 
> The conventional is shorter, and is conventional. Even the C++ Core Guidelines: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md
> continue to use the traditional syntax. Just wondering.

There are new capabilities in C++ that would only work with a trailing return syntax. But for returning an integer as your example shows, it makes no difference. 

One benefit why I like this syntax is that it allows your function declarations in the header look well aligned. E.g., every function name starts with auto and column 6 will be the start of the function name. (Suppose you put function implementations in the .cpp file.)

Reply:
> The _paintGL method is getting awfully long and complex. Can you make this 'if' clause into a helper function to improve readability?

This request is addressed in b75ef7ba .